### DISCUSSION TITLE
ArgoCD deployment dependencies between applications

### DISCUSSION THREAD

**Discussion by pcperera** [argoproj/argo-cd#8262] [Q&A] [answered]
Hi all,

We have below setup in our system which generates Grafana dashboards from code.

**Micro-service** 
This exposes an endpoint to fetch json for a given Grafana dashboard. This service is consumed by GrafanaDashboard manifest to generate the dashboard.

apiVersion: argoproj.io/v1alpha1
kind: Service
metadata:
  name: dashboard-service
spec:
  ports:
  - port: 8085
    name: http
  type: LoadBalancer
  selector:
    app: dashboard-service

**GrafanaDashboard manifests**
This uses the above micro service to fetch the json for the deployment of Grafana dashboard.

---
apiVersion: integreatly.org/v1alpha1
kind: GrafanaDashboard
metadata:
  name: some-dashboard
  labels:
    app: dev-grafana
spec:
  depends_on: dashboard-service
  customFolderName: dashboards
  url: http://dashboard-service:8080/dashboard/some-dashboard/json
  datasources:
    - inputName: "DS_PROMETHEUS"
      datasourceName: "Prometheus"

To modify the Grafana dashboard (json) we modify the micro service code. Grafana dashboard is however not automatically updated we we make changes to the micro-service source code repository.

Is it possible to create a dependency through ArgoCD to redeploy the Grafana dashboard kind every-time changes are made to the micro service repository ?

Thanks
Priyanka

**Comment by todaywasawesome** [comment, marked as answer]
The closest way I can think of doing what you describe would be to use a kustomization as part of your microservice repository that pulled in and generated the grafana dashboard and I would add a label with an explicitly set version that you would change as part of updates to your microservice. 

I believe in Grafana you can also use the [API to request dashboard updates](https://grafana.com/docs/grafana/latest/http_api/dashboard/). This could be added to a `PostSync` [hook](https://argo-cd.readthedocs.io/en/stable/user-guide/resource_hooks/). I think this approach would be infinitely better.
